contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+        address callbacks;    // should fulfill voting callbacks interface.
+        ProposalState state;
+        uint winningVote; //the winning vote.
+        address proposer;
+        uint currentBoostedVotePeriodLimit;
+        bytes32 paramsHash;
+        uint daoBountyRemain;
+        uint daoBounty;
+        uint totalStakes;// totalStakes[0] - (amount staked minus fee) - Total number of tokens staked which can be redeemable by stakers.
+                           // totalStakes[1] - (amount staked) - Total number of redeemable tokens.
+        uint confidenceThreshold;
+        uint expirationCallBountyPercentage;
+        uint[3] times; //times[0] - sumbmittedTime
+                       //times[1] - boostedPhaseTime
+                       //times[2] -preBoostedPhaseTime;
+        //      vote      reputation
+        mapping(uint    =>  uint     ) votes;
+        //      vote      reputation
+        mapping(uint    =>  uint     ) preBoostedVotes;
+        //      address     voter
+        mapping(address =>  Voter    ) voters;
+        //      vote        stakes
+        mapping(uint    =>  uint     ) stakes;
+        //      address  staker
+        mapping(address  => Staker   ) stakers;
+    }
+
+    event Stake(bytes32 indexed _proposalId, address indexed _organization, address indexed _staker,uint _vote,uint _amount);
+    event Redeem(bytes32 indexed _proposalId, address indexed _organization, address indexed _beneficiary,uint _amount);
+    event RedeemDaoBounty(bytes32 indexed _proposalId, address indexed _organization, address indexed _beneficiary,uint _amount);
+    event RedeemReputation(bytes32 indexed _proposalId, address indexed _organization, address indexed _beneficiary,uint _amount);
+    event GPExecuteProposal(bytes32 indexed _proposalId, ExecutionState _executionState);
+    event ExpirationCallBounty(bytes32 indexed _proposalId, address indexed _beneficiary,uint _amount);
+
+    mapping(bytes32=>Parameters) public parameters;  // A mapping from hashes to parameters
+    mapping(bytes32=>Proposal) public proposals; // Mapping from the ID of the proposal to the proposal itself.
+    mapping(bytes32=>uint) public orgBoostedProposalsCnt;
+           //organizationId => organization
+    mapping(bytes32        => address     ) public organizations;
+    //organizationId => averageBoostDownstakes
+    mapping(bytes32           => uint               ) public averagesBoostDownstakes;
+    uint constant public NUM_OF_CHOICES = 2;
+    uint constant public NO = 2;
+    uint constant public YES = 1;
+    uint public proposalsCnt; // Total number of proposals
+    StandardToken public stakingToken;
+    address constant GEN_TOKEN_ADDRESS = 0x543Ff227F64Aa17eA132Bf9886cAb5DB55DCAddf;
+
+    /**
+     * @dev Constructor
+     */
+    constructor(StandardToken _stakingToken) public
+    {
+      //The GEN token (staking token) address is hard coded in the contract by GEN_TOKEN_ADDRESS .
```

Comment:
because we want GEN token address to be hard coded in the protocol.
so when this contract will be deployed on any network (kovan,rinkeby,mainnet ) which host GEN token...the GEN token will be used.
This "hack" is simply to allow testing that using ganache.

Reply:
But it is not "hard coded in the protocol". Anyone can deploy this contract and not use the given token address.

Reply:
(Also: one address will only work on one network - you would need to be very careful to get the same addresses on rinkeby and kovan)

Reply:
yes. anyone can modify the contract and re deploy that...easy. though in that case it is not daostack's contract (with all means).
please also note that this GP contract is "universal" .can host many schemes .
GEN already have the same address on rinkeby, kovan and mainnet.

Reply:
**Nice** that they have the same address

I of course realize that you can rewrite the contract and depoy it. What I meant is that you can also use this exact same contract with another GEN token address, because you can alwas overrule the hard-coded address with a parameter. Anyway, code is fine :-)